### DISCUSSION TITLE
Is it possible to list the content of a group?

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#262] [Q&A] [answered]
I would like to run actions on each entity in a group. Right now I do it manually via

    for t in ['switch.michael_bureau', 'switch.michael_lampe_haute', 'switch.michael_lampe_mur']:

but I would like to make it more future-proof by doing something like 

    for t in get_group_entities('group.michael'):

Is this possible?

**Comment by dlashua** [comment, marked as answer]
The group will have an attribute called "entity_id" that is a List of all the entities in the group. You can use that.

```
for t in group.michael.entity_id:
   log.info(t)
```

**Reply by X52p** [reply]
How would I do this, if I just have a string containing the entity_id of the group? (and how could I check if it is a group?)

**Reply by dlashua** [reply]
```
somevar = "group.mine"
entities = state.getattr(somevar)['entity_id']
```

To check if "somevar" is actually a group, you can use regular python string methods to test.

```
if somevar.startswith('group.'):
  do a thing
```

**Reply by X52p** [reply]
Ah, I think I misunderstood the term "group" here. I did't know there were other types of groups than the ZAH groups.
I have a Zigbee Group of lights with the entity ID "light.ceilinglights_bedroom" and I would like to check what lights are in that group.

**Reply by dlashua** [reply]
Ah. I don't use ZHA so I'm not sure how those light groups work. You'll have to dig through the documentation and code for ZHA and see if you can figure it out. I use Zigbee2MQTT Light Groups and with those, from Home Assistant, a group of lights is not distinguishable from a single light. The look exactly the same and there's no way to tell.

You could always do it by convention, though. For your groups, always name the entities light.group_whatever. So you'll know if it says group in the entity id then it is. However, that still doesn't help you to know which lights are in the group.
